pkg/controller/instance/controller.go

```diff
@@ -135,6 +137,10 @@ func (c *Controller) Reconcile(ctx context.Context, req ctrl.Request) error {
 	// runtime object in it's fields.
 	rgRuntime, err := c.rgd.NewGraphRuntime(instance)
 	if err != nil {
+		// Mark graph resolution failure and update status before returning error
+		mark := NewConditionsMarkerFor(instance)
+		mark.GraphNotResolved("failed to create runtime resource graph definition: %v", err)
+		c.updateInstanceStatusOnError(ctx, instance)
```

Comment:
shall we propagate the error into the condition here?

Reply:
good call, let me look at this

Reply:
the condition already has the error, updateInstanceStatusOnError just manages the state field and updates conditions based on what is already in the condition manager.